### PR TITLE
Fix Securitydefinitions having wrong name

### LINKED ISSUE #616: Incorrect securityDefinitions generation
Upgrading from 3.1.5 to 3.1.7 results in incorrect header name in securityDefinitions. Using the following `securityDefinition.json`:
```
{
  "jwt": {
    "type": "apiKey",
    "name": "Authorization",
    "in": "header"
  }
}
```
generates this difference between 3.1.5 and 3.1.7 in swagger.json:
```
--- swagger.old 2018-05-07 08:49:39.260327034 +1000
+++ swagger.new 2018-05-07 08:50:44.195918130 +1000
@@ -14292,7 +14304,7 @@
   "securityDefinitions" : {
     "jwt" : {
       "type" : "apiKey",
-      "name" : "Authorization",
+      "name" : "jwt",
       "in" : "header"
     }
   },
```

### PR DESCRIPTION
**Pull request by Postremus**
This prevents the name field being used as overall name of the definition itself.
fixes #616

### CODE REVIEW

**Comment by Postremus** [comment]
This PR is now ready for review.

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/SecurityDefinition.java [hunk 1]
```diff
@@ -114,4 +145,12 @@ public String getIn() {
     public void setIn(String in) {
         this.in = in;
     }
+
+    public String getJson() {
```
Comment (Postremus) [inline src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/SecurityDefinition.java]: I need to remove these get / set methods I added. They may cause trouble somewhere else in smp.

Comment (Postremus) [reply src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/SecurityDefinition.java]: They are needed for the unit tests, so I will leave them here for now..

**Comment by Postremus** [comment]
This PR is now ready for review again.

**Comment by who** [comment]
Good work @Postremus . Merging now.

**Pull request merged**